src/ensembl/src/content/app/entity-viewer/gene-view/components/unspliced-transcript/UnsplicedTranscript.tsx

```diff
@@ -132,20 +124,14 @@ const Backbone = (
 
   for (let i = 0; i < spliced_exons.length; i++) {
     const {
-      relative_location: { start: exonStart, end: exonEnd }
+      relative_location: { end: exonEnd }
     } = spliced_exons[i];
-    if (i === 0) {
-      intervals.push([1, exonStart]);
-    } else {
-      const previousExon = spliced_exons[i - 1];
-      const {
-        relative_location: { end: previousExonEnd }
-      } = previousExon;
-      intervals.push([previousExonEnd, exonStart]);
-    }
 
-    if (i === spliced_exons.length - 1) {
-      intervals.push([exonEnd, transcriptLength]);
+    if (i < spliced_exons.length - 1) {
+      const {
+        relative_location: { start: nextExonStart }
+      } = spliced_exons[i + 1];
+      intervals.push([exonEnd, nextExonStart]);
     }
```

Comment:
I tried to simplify this a bit. I didn't see a difference between the transcript diagram with the old code and the new one, so perhaps this works. However, would like to know what you think about this.